pkg/service/restore_data.go

```diff
@@ -58,12 +60,21 @@ func (r *dataRestorer) Restore(request *model.RestoreRequestInternal) (RestoreJo
 		}
 		defer r.asClientCreator.Close(client)
 
-		restoreResult, err := r.restoreService.RestoreRun(ctx, client, request)
+		handler, err := r.restoreService.RestoreRun(ctx, client, request)
+		if err != nil {
+			r.restoreJobs.setFailed(jobID, fmt.Errorf("failed to start restore operation: %w", err))
+			return
+		}
+		r.restoreJobs.addTotalRecords(jobID, totalRecords)
+		r.restoreJobs.addHandler(jobID, handler)
+
+		// Wait for the restore operation to complete
+		err = handler.Wait(ctx)
```

Comment:
Update to the latest version of the library.

Reply:
wow, this update did helped fix the unit test, but caused a lot of diffs.